src/move.cpp

```diff
@@ -2035,6 +2036,11 @@ static void checkLocalFeatures(DROID *psDroid)
 
 		if (psObj->type == OBJ_FEATURE && !psObj->died)
 		{
+			if (std::abs(psDroid->pos.z - psObj->pos.z) > PICKUPZDIST)
```

Comment:
Please also add a comment about how this translates to tile coordinates (as noted in the PR description about half-a-tile difference).